### PR TITLE
feat: add PS/PSL for package metadata options to sfdx-project.json

### PR DESCRIPTION
**Pull request by KSCCO**
[@W-14842023@](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001iQVLhYAO/view)

Add support for additional properties in sfdx-project.json to allow specifying permission sets and permission set licenses to the build org user for package metadata deploy during the package version creation process.

### CODE REVIEW

File: src/package/packageVersionCreate.ts [hunk 1]
```diff
@@ -374,7 +374,7 @@ export class PackageVersionCreate {
       packageDescriptorJson = copyDescriptorProperties(packageDescriptorJson, definitionFileJson);
     }
```
Comment (KSCCO) [inline src/package/packageVersionCreate.ts:376]: A corresponding schema change to sfdx-project.json is required for these changes.

File: test/package/packageVersionCreate.test.ts [hunk 2]
```diff
@@ -593,6 +593,61 @@ describe('Package Version Create', () => {
     expect(package2DescriptorJson).to.have.string('buildOrgLanguage');
   });
```
Comment (KSCCO) [inline test/package/packageVersionCreate.test.ts:595]: In order to get this test working I had to modify node_modules/@salesforoce/core/lib/sfProject.d.ts and add  to PackageDir:
   packageMetadataAccess?:  {
       permissionSets: string | string[];
       permissionSetLicenses: string | string[];
    };

Is there a corresponding change that needs to be submitted for that PackageDir definition?

Comment (shetzel) [reply test/package/packageVersionCreate.test.ts:595]: Yes, we just need to update this library to use the version of `@salesforce/core` (aka sfdx-core) that has the schema and type definition updates.

File: src/package/packageVersionCreate.ts (hunk 1 above)
Comment (KSCCO) [reply src/package/packageVersionCreate.ts:376]: schema change has been merged: https://github.com/forcedotcom/schemas/commit/a1db98d229b8a6fd292f29a78acf7947027500f2

File: test/package/packageVersionCreate.test.ts (hunk 2 above)
Comment (KSCCO) [reply test/package/packageVersionCreate.test.ts:595]: The sfdx-core change has been merged: https://github.com/forcedotcom/sfdx-core/pull/1038

**Review by shetzel — APPROVED** [review]

**Pull request merged**